core/src/main/java/com/google/instrumentation/stats/BucketBoundaries.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2017, Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.instrumentation.stats;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * The optional histogram bucket boundaries for a {@link Distribution}.
+ */
+final class BucketBoundaries {
+  private final List<Double> bucketBoundaries;
+
+  /**
+   * @param bucketBoundaries the boundaries for the buckets in the underlying {@link Distribution}.
+   * @return a new {@code BucketBoundaries} with the specified boundaries.
+   * @throws NullPointerException if {@code bucketBoundaries} is null.
+   * @throws IllegalArgumentException if {@code bucketBoundaries} is not sorted, or has zero length.
+   */
+  static final BucketBoundaries create(List<Double> bucketBoundaries) {
+    checkNotNull(bucketBoundaries, "bucketBoundaries");
+    checkArgument(bucketBoundaries.size() > 0, "Zero length bucket boundaries");
+
+    List<Double> bucketBoundariesCopy = new ArrayList<Double>(bucketBoundaries.size());
```

Comment:
The list could be copied more efficiently with `List<Double> bucketBoundariesCopy = new ArrayList<Double>(bucketBoundaries);`.